arch/arm/core/aarch64/macro.inc

```diff
@@ -8,29 +8,6 @@
 #define _MACRO_H_
 
 #ifdef _ASMLANGUAGE
-/*
- * macro to support mov of immediate constant to 64 bit register
- * It will generate instruction sequence of 'mov'/ 'movz' and one
- * to three 'movk' depending on the immediate value.
- */
-.macro mov_imm, xreg, imm
-	.if ((\imm) == 0)
-		mov	\xreg, \imm
-	.else
-		.if (((\imm) >> 31) == 0 || ((\imm) >> 31) == 0x1ffffffff)
-			movz    \xreg, (\imm >> 16) & 0xffff, lsl 16
-		.else
-			.if (((\imm) >> 47) == 0 || ((\imm) >> 47) == 0x1ffff)
-				movz    \xreg, (\imm >> 32) & 0xffff, lsl 32
-			.else
-				movz    \xreg, (\imm >> 48) & 0xffff, lsl 48
-				movk    \xreg, (\imm >> 32) & 0xffff, lsl 32
-			.endif
-			movk    \xreg, (\imm >> 16) & 0xffff, lsl 16
-		.endif
-		movk    \xreg, (\imm) & 0xffff, lsl 0
-	.endif
-.endm
```

Comment:
If this macro needs to be used outside arch code (e.g. in SoC, driver, application code), I agree that this should be moved to a public header, preferably into `include/arch/arm/aarch64/macro.inc`.

Reply:
Having two files with `macro.inc ` name can be confusing.  How about  `asm.inc` .  I think it need not be strictly restricted to macros only.

Reply:
Personally I think `macro.inc` sounds better, but no strong opinion on this one.

Another possible alternative is to rename the existing private `macro.inc` to `macro_priv.inc`, as we normally do with private headers.

Reply:
ACK.   renamed `macro.inc  ->  macro_priv.inc`   and used `macro.inc  `for public macro for assembly code.